File: MainApp/src/main/java/com/sleticalboy/util/OperationUtils.java
package com.sleticalboy.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class OperationUtils {

    public static boolean isNumeric(String str) {
        Pattern pattern = Pattern.compile("[0-9]*");
        Matcher isNum = pattern.matcher(str);
        return isNum.matches();
    }

    public static String secretKey() {
        String resultStr = "";
        for (int i = 0; i < 16; i++) {
            int x = (int) (Math.random() * 95) + 32;
            char c = (char) x;
            resultStr += c;
        }
        return resultStr;
    }

    public static String convertKey(String inStr, char c) {

        char[] a = inStr.toCharArray();
        for (int i = 0; i < a.length; i++) {
            a[i] = (char) (a[i] ^ c);
        }
        return new String(a);
    }

}

File: app/src/main/java/com/binlee/learning/util/OperationUtils.kt
package com.binlee.learning.util

import java.util.regex.Pattern

object OperationUtils {

  fun isNumeric(str: String?): Boolean {
    val pattern = Pattern.compile("[0-9]*")
    val isNum = pattern.matcher(str)
    return isNum.matches()
  }

  fun secretKey(): String {
    var resultStr = ""
    for (i in 0..15) {
      val x = (Math.random() * 95).toInt() + 32
      val c = x.toChar()
      resultStr += c
    }
    return resultStr
  }

  fun convertKey(inStr: String?, c: Char): String {
    val a = inStr!!.toCharArray()
    for (i in inStr.toCharArray().indices) {
      a[i] = (a[i].code xor c.code).toChar()
    }
    return String(a)
  }
}